packages/server-core/src/user/login-token/login-token.hooks.ts

```diff
@@ -40,6 +46,17 @@ import {
   loginTokenResolver
 } from './login-token.resolvers'
 
+/**
+ * Generate token when creating a login token
+ * @param context
+ * @returns
+ */
+const generateToken = async (context: HookContext<LoginTokenService>) => {
+  const token = crypto.randomBytes(config.authentication.bearerToken.numBytes).toString('hex')
+
+  context.data = { ...context.data, token, expiresAt: toDateTimeSql(moment().utc().add(2, 'days').toDate()) }
```

Comment:
Both these things. token generation and expiresAt should be moved to data resolver